internal/installcfg/installcfg.go

```diff
@@ -271,6 +272,41 @@ func (i *installConfigBuilder) setImageContentSources(cfg *InstallerConfigBareme
 	return nil
 }
 
+func (i *installConfigBuilder) setVSpherelatformInstallconfig(cluster *common.Cluster, cfg *InstallerConfigBaremetal) {
+	vspherePlatform := cluster.Platform.Vsphere
+	if vspherePlatform == nil {
+		cfg.Platform = platform{
+			Vsphere: &platformVsphere{
+				APIVIP:     cluster.APIVip,
+				IngressVIP: cluster.IngressVip,
+			},
+			None:      nil,
+			Baremetal: nil,
+		}
+	} else {
+		p := platformVsphere{
+			VCenter:          *vspherePlatform.VCenter,
+			Username:         *vspherePlatform.Username,
+			Password:         *vspherePlatform.Password,
+			Datacenter:       *vspherePlatform.Datacenter,
+			DefaultDatastore: *vspherePlatform.DefaultDatastore,
+			Network:          *vspherePlatform.Network,
+			Cluster:          *vspherePlatform.Cluster,
+			APIVIP:           cluster.APIVip,
+			IngressVIP:       cluster.IngressVip,
+		}
+		if len(vspherePlatform.Folder) > 0 {
```

Comment:
it changed to 
``` 
if vspherePlatform.Folder != nil {
	p.Folder = *vspherePlatform.Folder
}
``` 
But we need it because Folder can be nil, and `p.Folder` is string, we have to verify `vspherePlatform.Folder` before assign it